src/main/java/com/yahoo/bullet/kafka/CertRefreshingSslEngineFactory.java

```diff
@@ -0,0 +1,205 @@
+        // SslChannelBuilder / SslFactory
+        return false;
+    }
+
+    @Override
+    public Set<String> reconfigurableConfigs() {
+        return Collections.emptySet();
+    }
+
+    @Override
+    public KeyStore keystore() {
+        // As of kafka-client 2.6 this method is only used on the server-side, so we don't need to return a real
+        // keystore, but we will anyway just in case future kafka-client versions use it on the client side as well.
+        try {
+            return createKeyStore(publicCertLocation, privateKeyLocation);
+        } catch (Exception e) {
+            log.error("Error creating keystore - this function should not be getting called on the client side.", e);
+            return null;
+        }
+    }
+
+    @Override
+    public KeyStore truststore() {
+        // As of kafka-client 2.6 this method is only used on the server-side, so we don't need to return a real
+        // truststore, but we will anyway just in case future kafka-client versions use it on the client side as well.
+        try {
+            return getKeyStore(truststoreLocation, truststorePassword.value().toCharArray());
+        } catch (Exception e) {
+            log.error("Error creating truststore - this function should not be getting called on the client side.", e);
+            return null;
+        }
+    }
+
+    @Override
+    public void close() throws IOException {
+        this.sslContext = null;
+    }
+
+    private SSLContext createSSLContext() {
+        try {
+            KeyRefresher keyRefresher = generateKeyRefresher(truststoreLocation,
+                                                             truststorePassword.value(),
+                                                             publicCertLocation,
+                                                             privateKeyLocation);
+            SSLContext sslContext = Utils.buildSSLContext(keyRefresher.getKeyManagerProxy(), keyRefresher.getTrustManagerProxy());
+            keyRefresher.startup(keyRefreshInterval);
+            log.info("Creating SSLContext that will refresh keys every {} ms.", keyRefreshInterval);
+            return sslContext;
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to create SSLContext.", e);
+        }
+    }
+
+    // Functions that proxy to other classes can be extended for testing
+
+    protected KeyStore getKeyStore(String jksFilePath, char[] password) throws Exception {
+        return Utils.getKeyStore(jksFilePath, password);
+    }
+
+    protected KeyStore createKeyStore(String athenzPublicCert, String athenzPrivateKey) throws Exception {
+        return Utils.createKeyStore(athenzPublicCert, athenzPrivateKey);
+    }
+
+    protected KeyRefresher generateKeyRefresher(String trustStorePath, String trustStorePassword, String athenzPublicCert, String athenzPrivateKey) throws Exception {
```

Comment:
Should add javadocs for all these protected methods to indicate what the override is for